src/components/TreebankService/TreebankService.js

```diff
@@ -0,0 +1,67 @@
+import React, { Component } from 'react';
+import { Redirect } from 'react-router-dom';
+import {
+  MessagingService,
+  ResponseMessage,
+  WindowIframeDestination as Destination,
+} from 'alpheios-messaging/dist/dev/alpheios-messaging';
+
+const config = {
+  name: 'treebank',
+  targetIframeID: 'string-not-used',
+  targetURL: 'string-not-used',
+};
+
+const error = (request, message) => ResponseMessage.Error(request, new Error(message));
+
+class TreebankService extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = { redirectTo: null };
+
+    this.service = new MessagingService('treebank-service', new Destination(config));
+
+    this.messageHandler = this.messageHandler.bind(this);
+  }
+
+  componentDidMount() {
+    const { name } = config;
+
+    // TODO this line is causing the error (even with try { }):
+    // TypeError: Cannot set property 'origin' of undefined
+    this.service.registerReceiverCallback(name, this.messageHandler);
+  }
+
+  componentWillUnmount() {
+    // TODO is there any way to unregister in alpheios-messaging?
+    // this.service.unregisterReceiverCallback(name, this.messageHandler);
+  }
+
+  messageHandler(request, responseFn) {
+    const { body } = request;
+    const [name] = Object.keys(body);
+
+    switch (name) {
+      case 'gotoSentence':
```

Comment:
What other API calls will you need for the integration with Alpheios? Will you need `prevSentence` and `nextSentence`?

The `refresh` and `getMorph` calls should be easy to implement once we get the newer version of Alpheios working with the Treebank Template.

Reply:
I think I just need goToSentence, refresh and getMorph.